File: src/main/java/de/tectoast/emolga/commands/draft/StopdraftCommand.java
package de.tectoast.emolga.commands.draft;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.draft.Draft;

public class StopdraftCommand extends Command {
    public StopdraftCommand() {
        super("stopdraft", "Beendet den Draft", CommandCategory.Flo);
        setArgumentTemplate(ArgumentManagerTemplate.builder().add("draftname", "Draftname", "Der Name des Drafts", ArgumentManagerTemplate.draft())
                .setExample("!stopdraft Emolga-Conference")
                .build());
    }

    @Override
    public void process(GuildCommandEvent e) {
        if (Draft.drafts.removeIf(d -> {
            if (d.name.equals(e.getArguments().getText("draftname"))) {
                d.cooldown.cancel(false);
                return true;
            }
            return false;
        })) {
            e.reply("Dieser Draft wurde beendet!");
        } else {
            e.reply("Dieser Draft existiert nicht!");
        }
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/draft/StopdraftCommand.kt
package de.tectoast.emolga.commands.draft

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.json.db
import de.tectoast.emolga.utils.json.emolga.draft.League
import de.tectoast.emolga.utils.json.eq
import de.tectoast.emolga.utils.json.findOne
import kotlinx.coroutines.cancel

class StopdraftCommand : Command("stopdraft", "Beendet den Draft", CommandCategory.Flo) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder()
            .add("draftchannel", "Draftchannel", "Der Channel des Drafts", ArgumentManagerTemplate.DiscordType.CHANNEL)
            .setExample("!stopdraft #stufe1-draft")
            .build()
    }

    override suspend fun process(e: GuildCommandEvent) {
        e.reply(
            db.drafts.findOne(League::tcid eq e.arguments.getChannel("draftchannel").idLong)?.run {
                cooldownJob!!.cancel("Draft stopped by command")
                isRunning = false
                save()
                "Der Draft wurde beendet!"
            } ?: "In diesem Channel läuft derzeit kein Draft!"
        )
    }
}
